### PR TITLE
Apply Nudge Chips Policy #2086

### PR DESCRIPTION
**Pull request by shshshl09**
* change ASR's state from enum to sealed class
* Not render nudge chips if ASR(DM) not initiated by ExpectSpeech directive.

### CODE REVIEW

**Comment by shshshl09** [comment]
https://tde.sktelecom.com/pms/browse/NUGUSDKQA-225?filter=-1

**Review by tommy5oo — APPROVED** [review]

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt [hunk 1]
```diff
@@ -24,42 +24,50 @@ import com.skt.nugu.sdk.core.interfaces.message.Header
...
-        EXPECTING_SPEECH,
+        object EXPECTING_SPEECH : State() {
+            override val name: String = "EXPECTING_SPEECH"
+        }
 
         /**
          * Recognizing: speech recognition started, but not speech started yet.
          */
-        LISTENING,
+        class LISTENING(val initiator: Initiator) : State() {
```
Comment (Yuseon-Han) [inline nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt:47]: Every listening makes LISTENING object and it isn't efficient. How about make LISTENING Object also.

Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt:47]: How can I do it?

Comment (Yuseon-Han) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt]: How about these below,

```
object LISTENING : State() {
            var initiator: Initiator? = null // nullable or default value if it exist
            override val name: String = "LISTENING"
        }
```


and..in DefaultAsrAgent.onStateChanged()

`ASRAgentInterface.State.LISTENING.also { it.initiator = currentRequest?.second?.initiator ?: ASRAgentInterface.Initiator.WAKE_UP_WORD }`

Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt]: It is not safe sharing mutable object.

Comment (Yuseon-Han) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt]: That's right. then how about this below

```
object LISTENING: State() {
            var initiator: Initiator? = null
                private set

            fun withInitiator(i : Initiator) : LISTENING{
                initiator = i
                return this
            }

            override val name: String = "LISTENING"
        }
```

// DefaultAsrAgent.onStateChanged()
```
 ASRAgentInterface.State.LISTENING.withInitiator(currentRequest?.second?.initiator ?: ASRAgentInterface.Initiator.WAKE_UP_WORD)
```

This code also dosen't need to change comparison using 'is'.

Comment (Yuseon-Han) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt]: Yes that's what I intended.
The State is only one. 'LISTENING' is same as other State like IDLE, BUSY. It's only one. The different thing is initiator.
So I thought State not need to be newly created. And also It's better for memory and Client work.

**Pull request merged**